opm/output/eclipse/WindowedArray.hpp

```diff
@@ -0,0 +1,177 @@
+/*
+  Copyright (c) 2018 Statoil ASA
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef OPM_WINDOWED_VECTOR_HPP
+#define OPM_WINDOWED_VECTOR_HPP
+
+#include <cassert>
+#include <iterator>
+#include <type_traits>
+#include <vector>
+
+#include <boost/range/iterator_range.hpp>
+
+namespace Opm { namespace RestartIO { namespace Helpers {
+
+    template <typename T>
+    class WindowedArray
+    {
+    private:
+        using Vector = std::vector<T>;
+
+    public:
+        using WriteWindow = boost::iterator_range<
+            typename Vector::iterator>;
+
+        using ReadWindow = boost::iterator_range<
+            typename Vector::const_iterator>;
+
+        using Idx = typename Vector::size_type;
+
+        struct NumWindows { Idx value; };
```

Comment:
Ehhh - this looks a bit over-engineered to me.  The literal `Idx` - or something very similar - is used many places in the OPM code base as a variable name; I think it is confusing that it is now used as a type.

Reply:
> Ehhh - this looks a bit over-engineered to me.

Reduces the risk of mixing number of windows with window size.

Reply:
I would like if `Idx` was renamed `IndexType` or something - the string `Idx` looks like an instance and not a type.